Automattic/wp-calypso
# languages: Use languages data keys for LanguageSlug type def

#### Changes proposed in this Pull Request

* Improve `LanguageSlug` type def to use union of allowed language slugs.
* Export `LanguageSlug` type.
* Update `@automattic/language-picker` to use Language typedef from `@automattic/languages`.

#### Testing instructions

* Code review.
* Confirm `@automattic/languages` and `@automattic/language-picker` packages builds properly:
  * `cd packages/languages && yarn build`
  * `cd packages/language-picker && yarn build`
